examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py

```diff
@@ -46,14 +46,23 @@ def virtual_step(self, train_x, train_y, xi, w_optim):
         # Compute gradient
         gradients = torch.autograd.grad(loss, self.model.getWeights())
 
+        # Check device use cuda or cpu 
+        use_cuda = list(range(torch.cuda.device_count()))
+        if use_cuda:
+            print("Using CUDA")
+        device = torch.device("cuda" if use_cuda else "cpu")
+        
```

Comment:
We identify device here: https://github.com/kubeflow/katib/blob/4d3ea0c639da99848ee9409c8e91122874c306a5/examples/v1beta1/trial-images/darts-cnn-cifar10/run_trial.py#L86-L100.
Can we just pass the device to the [Architect](https://github.com/kubeflow/katib/blob/4d3ea0c639da99848ee9409c8e91122874c306a5/examples/v1beta1/trial-images/darts-cnn-cifar10/run_trial.py#L143) class ?

Reply:
Yes, we can. But is it a good idea to send the device name?

Reply:
I think, it's fine since we don't need to invoke `torch` API again to understand if we have GPU available.

Reply:
OK I will change it.

Reply:
@andreyvelich Please check it and thanks your help.

Reply:
Thanks, I restarted tests.